Add an SNS topic for CloudWatch alarms

jsf9k opened the pull request:
## 🗣 Description ##

This pull request:
- Makes use of the AWS Organizations read-only provider to obtain the ID of the "Users" AWS account.  Previously the ID of the "Users" account was being passed in as an input parameter.
- Makes use of [cisagov/cw-alarm-sns-tf-module](https://github.com/cisagov/cw-alarm-sns-tf-module) to create an SNS topic that can be published to in order to send a message to the account email when a CloudWatch alarm is triggered.

## 💭 Motivation and context ##

- Passing in the ID of the "Users" account explicitly has been needing to be cleaned up for years.
- We require such an SNS topic so that we can receive emails when CloudWatch alarms indicate a potential problem with the COOL.

## 🧪 Testing ##

All automated testing passes.  This code has already been applied to:
- All non-dynamic accounts
- All dynamic staging accounts

A script is currently running to apply the changes to all dynamic production accounts.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert dependencies to default branches.

jsf9k committed:
Use the Organizations read-only provider vice passing in the Users account ID

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k committed:
Add @mcdonnnj as a codeowner

dav3r commented on shared-services/locals.tf:
```diff
@@ -0,0 +1,15 @@
+  users_account_id = [
+    for account in data.aws_organizations_organization.cool.accounts :
+    account.id
+    if account.name == "Users" && length(regexall("2020", account.email)) > 0
```
We no longer have the old "Users" account hanging around (it has since been deleted), so unless anyone has objections, we can just do this now:
```suggestion
    if account.name == "Users" 
```

jsf9k replied:
Please see commit 23f49e4.

dav3r replied:
Excellent 👍

dav3r reviewed (approved):
I have one suggestion (which probably needs to be applied elsewhere as well), but other than that, this all makes sense to me. 👍

jsf9k committed:
Simplify code to locate the Users account

The old, defunct Users account is no longer around; therefore, there
is no need to keep including the logic clause that checks the account
email.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj commented on shared-services/providers.tf:
```diff
@@ -12,3 +12,13 @@ provider "aws" {
+# Read-only AWS Organizations provider
+provider "aws" {
+  alias = "organizationsreadonly"
+  default_tags {
+    tags = var.tags
+  }
```
Just for my curiosity: If this is a read-only provider does setting tags do anything?

jsf9k replied:
It doesn't do anything, but I think it's a good idea to have the habit of always applying tags to the providers so that they always flow down to any resources that are created.

mcdonnnj commented on shared-services/locals.tf:
```diff
@@ -0,0 +1,15 @@
+# Retrieve the information for all accounts in the organization.  This
+# is used, for instance, to lookup the account IDs for the user
```
```suggestion
# is used, for instance, to lookup the account ID for the Users
```

mcdonnnj reviewed (approved):
Strong work and I'm looking forward to this functionality. I had one minor question (feel free to resolve it if I miss it and it's holding you up) and a documentation update suggestion for consideration.

jsf9k committed:
Correct number and capitalization in a comment

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Revert to the default branch of cisagov/cw-alarm-sns-tf-module

There is no need to use the first-commits branch now that
cisagov/cw-alarm-sns-tf-module#5 has been approved and merged.

jsf9k committed:
Use the Organizations read-only provider vice passing in the Users account ID

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k committed:
Use the Organizations read-only provider vice passing in the Users account ID

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k committed:
Use the Organizations read-only provider vice passing in the Users account ID

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k committed:
Use the Organizations read-only provider vice passing in the Users account ID

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k committed:
Simplify code to locate the Users account

The old, defunct Users account is no longer around; therefore, there
is no need to keep including the logic clause that checks the account
email.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Use the Organizations read-only provider vice passing in the Users account ID

jsf9k commented:
There are ten other COOL AWS accounts where this jazz has not been applied:
- `Cracking`
- `Domain Manager` (prod and staging)
- `inl0`
- `PCA` (prod and staging)
- `Playground Legacy`
- `RTA Legacy`
- `User Services` (prod and staging)

@dav3r, do we want to apply this jazz to any of those accounts other than `User Services`?

dav3r commented on audit/README.md:
```diff
@@ -32,16 +32,16 @@ To do this, follow these steps:
+   tags = {
+     Team        = "VM Fusion - Development"
+     Application = "COOL - Shared Services Account"
+     Workspace   = "staging"
```
Why do some of these README examples have `Workspace   = "staging"` and others have `Workspace   = "production"?`. Shouldn't we just make them all the same?

jsf9k replied:
Most of these accounts don't have separate staging and production versions.

dav3r replied:
So then maybe we should just make them all consistent with `Workspace = "production"`?

jsf9k replied:
I hope you choke on commit 39a66ae sir!

dav3r replied:
Thank you for your support. 😝

dav3r commented on audit/README.md:
```diff
@@ -32,16 +32,16 @@ To do this, follow these steps:
-   ]
+   tags = {
+     Team        = "VM Fusion - Development"
+     Application = "COOL - Shared Services Account"
```
```suggestion
     Application = "COOL - Audit Account"
```

dav3r reviewed (approved):
Looks great! 👍   Please see my two small documentation comments.

dav3r commented:
> @dav3r, do we want to apply this jazz to any of those accounts other than `User Services`?

We should probably do this in the `Domain Manager` and `PCA` accounts as well.

jsf9k committed:
Correct account name in example code

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj reviewed (changes requested):
Strong works 💪💪💪 The one thing I'd like to ask is that the configuration in `dynamic/` should also be updated to determine the `users_account_id` instead of taking it as a required variable.

jsf9k committed:
Add an SNS topic for CloudWatch alarms

jsf9k commented:
> Strong works musclemusclemuscle The one thing I'd like to ask is that the configuration in `dynamic/` should also be updated to determine the `users_account_id` instead of taking it as a required variable.

Ah yes.  I neglected to push after testing my changes in that directory.  Thanks for catching that.

jsf9k committed:
Specify the production workspace in all README tag examples

This is done (begrudgingly, on my part) so that all the READMEs are
gloriously consistent.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r commented on dynamic/README.md:
```diff
@@ -43,7 +43,13 @@ To do this, follow these steps:
-   users_account_id     = "222222222222"
+
+   tags = {
+     Application       = "COOL - env0 Staging Account"
```
Don't hate me:
```suggestion
     Application       = "COOL - env0 Production Account"
```

jsf9k committed:
Modify one last instance of "staging" in example tags

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
👍 👍

mcdonnnj reviewed (approved):
:shipit:
